test/gtest/ucm/malloc_hook.cc

```diff
@@ -506,17 +506,20 @@ UCS_TEST_F(malloc_hook_cplusplus, mmmap_ptrs) {
     set();
 
     const size_t   size    = ucm_dlmallopt_get(M_MMAP_THRESHOLD) * 2;
-    const size_t   max_mem = ucs_min(ucs_get_phys_mem_size() / 4, 4 * UCS_GBYTE);
+    const size_t   max_mem = ucs_min(ucs_get_phys_mem_size() / 8, 4 * UCS_GBYTE);
     const unsigned count   = ucs_min(400000ul, max_mem / size);
     const unsigned iters   = 100000;
 
     std::vector<std::string> ptrs;
 
     size_t large_blocks = 0;
 
-    /* Allocate until we get MMAP event */
+    /* Allocate until we get MMAP event
+     * Lock memory to avoid going to swap and ensure consistet test results.
+     */
     while (m_mapped_size == 0) {
         std::string str(size, 'r');
```

Comment:
it's better to use std::vector here because std::string allocates at least +1 for null-terminator.